### PR TITLE
MorphOS Platform Support

### PR DESCRIPTION
**Pull request by sev-**
This is the rebased version of #1624 where I accidentally pushed a clean master and that closed the previous PR...

### CODE REVIEW

**Comment by sev-** [comment]
@BeWorld2018 Could you please check that it still compiles nicely and if you notice any oddities, please tell me.

I rebase the whole thing.

**Comment by sev-** [comment]
Your previous master lives in `mos` branch

File: configure [hunk 1]
```diff
@@ -3435,6 +3455,28 @@ if test -n "$_host"; then
...
+
+
+			append_var DEFINES "-DDISABLE_SID"
+			append_var DEFINES "-DDISABLE_NES_APU"
+			append_var CXXFLAGS "-fno-exceptions"
+			append_var CXXFLAGS "-fno-rtti"
+			_backend="ps2"
+			_build_scalers=no
+			_mt32emu=no
+			;;
```
Comment (ccawley2011) [inline configure:3479]: This appears to be a leftover from the PS2 port, and is causing compilation to fail.

```suggestion
```

File: backends/fs/stdiostream.cpp [hunk 2]
```diff
@@ -87,6 +87,25 @@ bool StdioStream::flush() {
...
+		setvbuf(handle, NULL, _IOFBF, 8192);
+	}
+#endif
+
+#if defined(__WII__)
+	// disable newlib's buffering, the device libraries handle caching
+	if (handle)
+		setvbuf(handle, NULL, _IONBF, 0);
+#endif
+
```
Comment (ccawley2011) [inline backends/fs/stdiostream.cpp:108]: The buffer size is already set in `MorphOSFilesystemNode::createReadStream()`, meaning that this isn't needed any more.

```suggestion
```

File: backends/graphics/opengl/opengl-sys.h [hunk 3]
```diff
@@ -30,6 +30,10 @@
 #include "backends/platform/sdl/sdl-sys.h"
 #endif
 
+#ifdef __MORPHOS__
+#define USE_GLES_MODE 1
+#endif
+
```
Comment (ccawley2011) [inline backends/graphics/opengl/opengl-sys.h:36]: Wouldn't it be better to add `_opengl_mode=gles` to configure instead?

File: backends/platform/sdl/sdl-window.cpp [hunk 4]
```diff
@@ -203,7 +203,11 @@ bool SdlWindow::getSDLWMInformation(SDL_SysWMinfo *info) const {
 #if SDL_VERSION_ATLEAST(2, 0, 0)
 	return _window ? (SDL_GetWindowWMInfo(_window, info) == SDL_TRUE) : false;
 #else
+	#ifdef __MORPHOS__
+	return -1;
+	#else
 	return SDL_GetWMInfo(info);
+	#endif
```
Comment (ccawley2011) [inline backends/platform/sdl/sdl-window.cpp:210]:
```suggestion
#elif !defined(__MORPHOS__)
	return SDL_GetWMInfo(info);
#else
	return false;
```

File: backends/graphics/opengl/opengl-sys.h (hunk 3 above)
Comment (BeWorld2018) [reply backends/graphics/opengl/opengl-sys.h:36]: mmm, no sur because MorphOS have only old OpenGL 1.2 (tinygl) so GLES....., need to test unless this

Comment (BeWorld2018) [reply backends/graphics/opengl/opengl-sys.h:36]: I prepare all to test and rebuild it.

Comment (BeWorld2018) [reply backends/graphics/opengl/opengl-sys.h]: Ok i could built it (2 hours with my iMac PPC 2.1Ghz....) ! i remove USE_GLES_MODE

**Comment by sev-** [comment]
Okay, time to merge. Will you be able to provide builds for the upcoming release?

**Comment by BeWorld2018** [comment]
Yes no problem, i can  rebuild a release version

**Comment by sev-** [comment]
@BeWorld2018 hey hey! The release will be announced tomorrow. Would it be possible to provide the build?

**Pull request merged**